tests/unit/configuration_test.py

```diff
@@ -2376,3 +2389,45 @@ def test___verify_or_set_optional_string_with_valid_values(self):
             config_description=None,
             valid_values=["bar", "baz"],
         )
+
+    def test_max_send_rate_enforcement_legacy_defaults(self):
+        self._write_file_with_separator_conversion(
+            """ {
+                api_key: "hi",
+                max_send_rate_enforcement: "legacy"
+            }
+            """
+        )
+        config = self.get_configuration()
+        config.parse()
+
+        self.assertEquals(config.max_send_rate_enforcement, "legacy")
+        self.assertIsNone(config.parsed_max_send_rate_enforcement)
+
+        self.assertEquals(config.max_allowed_request_size, 1048576)
+        self.assertEquals(config.pipeline_threshold, 1.1)
+        self.assertEquals(config.min_request_spacing_interval, 1.0)
+        self.assertEquals(config.max_request_spacing_interval, 5.0)
+        self.assertEquals(config.max_log_offset_size, 5242880)
+        self.assertEquals(config.max_existing_log_offset_size, 104857600)
+
+    def test_disable_max_send_rate_enforcement_overrides(self):
```

Comment:
Also add a test case that verifies that if `max_send_rate_enforcement` is set to `unlimited` and you have values set in the configuration file for all of these parameters, then you still get the unlimited settings.